haskell/test/SMT/Types.hs

```diff
@@ -0,0 +1,141 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module SMT.Types
```

Comment:
It would be a lot better for understanding how this works to have Haddock comments on all the types and functions in here IMO.

Reply:
Added the Haddock comments.